### PR TITLE
Sepia slime extract works like a grenade [WIP]

### PR DESCRIPTION
**Pull request by 4dplanner**
:cl:
add: sepia slime extract has a delay before activating
balance: the user of a sepia slime extract is no longer immune to the time field
balance: chilling sepia is now the support extract - it always freezes the user, but not other marked people
add: burning sepia spawns the rewind camera. Take a selfie with someone and give it to them to make sure they remember the moment forever!
add: recurring sepia recalls to the hand of the last person to touch it after activating
/:cl:

Why this is a good thing: sepia slime extract is very overpowered right now. You can spam it, it's instant and unreactable, and has no risk to the user (unlike similar aoe threats, such as grenades). 

This should nerf it while adding some interesting play such as lobbing through windows, and still leaving it very strong.

todo:
- [x] recurring sepia
- [ ] cameras
- [ ] purple sepia
- [ ] dark blue sepia
- [ ] a solution for holymelon abuse?
- [ ] a better way for recurring sepia to work

### CODE REVIEW

**Comment by Shadowflame909** [comment]
So now it freezes you. Does it Still do that if you go near the time clock? Or just the initial reaction

**Comment by imsxz** [comment]
redpilled

**Comment by 4dplanner** [comment]
So the way this works will be:
t=0: sepia slime is injected
You then throw it
t=5: it spawns the big clock
Anyone walking into the big clock is frozen as usual, including you.

**Comment by Shadowflame909** [comment]
Oh. So this won't be very useful outside of a ranged laser attack.

**Comment by iskyp** [comment]
yeah but we already have large grenade casings 
why not just make it freeze the user

**Comment by 4dplanner** [comment]
The changelog already says that
balance: sepia slime extract freezes the user if they're in the AOE

**Comment by 4dplanner** [comment]
It'll still be useful for several reasons - 
1:if you hit someone with it they are basically dead, because you can just throwing weapon them to death/stand just outside
2: powerful area denial/escape ability
3: defensive purposes

**Comment by Shadowflame909** [comment]
Kind of misleading as that's not the important feature

**Comment by 4dplanner** [comment]
There are two, are you serious?

**Comment by iskyp** [comment]
hence why not just make it freeze the user instead of making it also have a delay
this is a stealth nerf to the large grenade casing if you want to get big brained about it

**Comment by 4dplanner** [comment]
If you just make it freeze the user it'd be useless, why would you freeze yourself?

**Comment by 4dplanner** [comment]
You shouldn't HAVE to use the large casing to make it worthwhile

**Comment by AdamElTablawy** [comment]
making it freeze the user would make it useless, tbh. the delay before activation should be strong enough-it's very powerful, but still.

also, this would make recurring slime extracts worse than useless-you'd activate it and then someone could pick it up, activate it and throw it back *to you.*

crossbreeds shouldn't be affected by this, basically, and a delay in activation SHOULD be enough to make this not disgustingly powerful anymore. even if you're not going to bother with the removing the freeze the user effect, don't change crossbreeds for this-they're time consuming enough to deserve a reward and the only crossbreed that really needs a nerf is the permanent methspeed slime

(also i'm fairly sure holy melons and null rods block slime timestop magic regardless).

**Comment by 4dplanner** [comment]
I don't believe that "it takes a long time" is justification to have that level of power.

**Comment by 4dplanner** [comment]
As to just having the delay, that would make spam throwing and also just running at people far too good

**Comment by iskyp** [comment]
the delay is stupid because, if you don't want to freeze yourself, you should make it into a grenade
>you shouldn't have to use a grenade to make it worthwhile

then why have large grenade casings when their entire point is to be grenades for slime extracts

**Comment by 4dplanner** [comment]
Just because one specific slime extract doesn't need them, doesn't mean they're not useful for any of the others

**Comment by 4dplanner** [comment]
Grenades should be an optional upgrade for a slime extract, not a necessary function for them.

**Comment by 4dplanner** [comment]
rephrased to the shadow flame way

**Comment by iskyp** [comment]
i think it should need them though which is my entire argument, because having no delay while still freezing the user would balance them in exactly the same way you're trying to achieve here just with a little more hassle to achieve breaking the flow of time itself

**Comment by AdamElTablawy** [comment]
While I don't like to use other things as an example of being overpowered as a justification for another thing, I could point out that *most* things scale with power like that. Mechs, for one, or genetics.

Still, that's not necessarily a good example, so here's a better justification-Each crossbred chilling slime can only be used once ever since I fixed it, and recurring slimes can be used against you, accordingly, and neither can be mass produced. You can either create a large amount of sepia slimes-which you already do-and attempt to stunlock someone down with timestops, and while that's viable it's not an incredibly effective tactic. It's certainly one of the strongest tools xenobiology has by default, alongside mobs and instant heals and all of that, and with the old infinite timestop being gone you now need to actually stock up on plasma and slime cores for these things to function-which takes a fair amount of effort, considering that slimes are obnoxious to breed and take a while and so on and so forth. Reproductive extracts neutralize some of that, but still.

You're basically turning one extract from a very good extract which can be used as a weapon of mass destruction via timestop chaining someone to a weapon that can't really be used very well at all since it freezes you too and has a smaller radius than a chemical grenade which you could make in science regardless. There's a reason nobody uses oil slimes.

The delay alone should be a nerf to their function, and punishing someone for going to the effort to get a crossbred slime, intentionally meant to be fairly powerful for all the time it takes to breed ten cerulean slime cores and slot them into an adult sepia slime is kinda shitty.

**Comment by 4dplanner** [comment]
You're right, it would work. I guess I just think that'd be a bit of an over-nerf before we've seen how this plays out.

**Comment by 4dplanner** [comment]
So there are two parts here - firstly, I don't think it's THAT much effort. Make a reproductive cerulean, reap the rewards. 

Secondly, even having only say 5 chilling extracts would be far too good.

As to the use of the new sepias, I still think they'd be very useful. As I've said, they confirm a kill, and what makes them better than mere explosions is the fact the field stays behind. Recurring slimes would admittedly have a pretty strong nerf - but how many people you'd be fighting just happen to carry plasma syringes around?

**Comment by AdamElTablawy** [comment]
I like the idea of being able to lob a slime core at someone for a timestop, and I like the idea of cerulean slimes having some backlash, since someone could activate the slime and throw it back at you if you use a recurring to get around inventory management problems, but the freezing the user makes it a shitty idea to use in general because it's really easy to kill yourself with in the process as a result. 

Chilling extracts do not freeze the user anyway, because you can mark yourself as not to be frozen regardless and thus would still be superior to ordinary slime extracts for the exact same reason you are concerned they would be 'too good.' What you just said doesn't make much sense on that mark.

**Comment by 4dplanner** [comment]
It's WIP, I haven't decided on what to do with chilling slimes yet.

**Comment by 4dplanner** [comment]
I agree that if just this was merged it'd be pretty inconsistent.

**Comment by Shadowflame909** [comment]
Wait you can put slime extracts in grenades? Oh shit. Why don't I see orange mini fireball and sepia slime grenades

**Comment by AdamElTablawy** [comment]
Crossbreeding can usually take more time than the round actually HAS, because of RNG and murderbone interrupting you as well as how annoying it can be to get the necessary things for reproductive extracts, etc.

@Shadowflame909 Because phlog grenades are easier to make, which is why orange slimes suck, and sepias can just be popped normally.

**Comment by Shardj** [comment]
Feels like chilling sepia is now in the right place, since to replicate the functionality of what was recurring sepia with chilling sepia you need 10u plasma each time. What I don't like is how recurring sepia is now just useless. I think sepia could be nerfed in better ways, simply shortening the freeze time would be a more appropriate nerf. I'm sure there are other more appropriate nerfs too

**Comment by Tlaltecuhtli** [comment]
you just need to apply brain think to make the recurring ones work

**Comment by 4dplanner** [comment]
I'd honestly rather snowflake recurring sepia than sabotage what I believe to be otherwise a good nerf. Would you say other than recurring it was fine?

As to the plasma part, plasma dust is too easily acquired to be a good balancing lever

**Comment by ExcessiveUseOfCobblestone** [comment]
Train came early today.

File: code/modules/reagents/chemistry/recipes/slime_extracts.dm [hunk 1]
```diff
@@ -559,9 +559,9 @@
 	required_other = TRUE
 
 /datum/chemical_reaction/slime/slimestop/on_reaction(datum/reagents/holder)
+	sleep(50)
 	var/turf/T = get_turf(holder.my_atom)
-	var/list/M = list(get_mob_by_key(holder.my_atom.fingerprintslast))
-	new /obj/effect/timestop(T, null, null, M)
+	new /obj/effect/timestop(T, null, null, null)
```
Comment (nichlas0010) [inline code/modules/reagents/chemistry/recipes/slime_extracts.dm]:
```suggestion
	new /obj/effect/timestop(T)
```

**Review by Amelia0010 — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/recipes/slime_extracts.dm (hunk 1 above)
Comment (4dplanner) [reply code/modules/reagents/chemistry/recipes/slime_extracts.dm]: ...good point

**Comment by optimumtact** [comment]
4dplanner making an an enemy of sci mains next

**Comment by nichlas0010** [comment]
why...

**Comment by 4dplanner** [comment]
I'm burned out for the moment honestly

**Comment by FrostyFridge** [comment]
thank god

**Comment by Shadowflame909** [comment]
Thus the tale of a man fighting against the salt of the playerbase comes to a halt. He needed a break from being viewed as the bad guy. But surely, another one to burden these negative feelings shall arise...

**Comment by imsxz** [comment]
i cant believe xenobabies killed our hero

**Comment by Whoneedspacee** [comment]
@4dplanner take a good break and honestly just work on whatever you're passionate about. I'm honestly feeling the same way as you.

**Comment by optimumtact** [comment]
I didn't say you could sleep

**Comment by KomradeSpectre** [comment]
Downvoted because I think the better way of balancing the sepia extract/time stopping is to make the field slow users and projectiles down instead of completely stopping them. Making the spell be "Slow Time" instead of "Time Stop". This would allow counterplay, but still have good strength to it.

**Comment by Shardj** [comment]
> Downvoted because I think the better way of balancing the sepia extract/time stopping is to make the field slow users and projectiles down instead of completely stopping them. Making the spell be "Slow Time" instead of "Time Stop". This would allow counterplay, but still have good strength to it.

That would make it useless

**Comment by zeroisthebiggay** [comment]
You know this is like the only 4dplanner nerf I partially agree with.
I remember a time I was a Nightmare, a scientist ran up to me _in maintenance_, stopped me with a sepia extract and proceeded to beat me to death with a crowbar while they shined a flashlight on me.
It should not be possible to fight a side antagonist with one of their hard counters being "bring a friend" in a one on one fight with **a fucking crowbar**.

Though I think it'd probably better to rework it into a slow time rather than a full _"congratulations you're stunlocked as a nukie and this scientist is going to slowly kill you with a toolbox"_.

**Comment by 4dplanner** [comment]
So you agree when u are the one who is ded :^)


I still might well finish this later, though.

**Comment by Shardj** [comment]
> You know this is like the only 4dplanner nerf I partially agree with.
> I remember a time I was a Nightmare, a scientist ran up to me _in maintenance_, stopped me with a sepia extract and proceeded to beat me to death with a crowbar while they shined a flashlight on me.
> It should not be possible to fight a side antagonist that excells in one on one combat (with one of their hard counters being "bring a friend") in a head on fight with **a fucking crowbar**. I ded.
> 
> Though I think it'd probably better to rework it into a slow time rather than a full _"congratulations you're stunlocked as a nukie and this scientist is going to slowly kill you with a toolbox"_.

Just stun them before they do that... you had all of the advantages still, that's like complaining that you were hit with a tazer even though you have a hitscan rocket launcher

**Comment by 4dplanner** [comment]
You can't really use melee stuns against a scientist with a 5x5 melee instakill

**Comment by Shardj** [comment]
Been talking about balancing this a lot in discord, main problem is that you can effectively carry tons of these making them disproportionately strong for how easy they are to get. A single use sepia would be less deadly than a grenade, and you have to work far harder to get it than you would to even get a maxcap. It's the botany problem again, the sheer volume of a certain strong weapon you can carry thanks to department plant/bio bags. Spamming timestops, making you effectively unstoppable is the issue here.

So the **proposed solution** which we like as a basepoint nerf to try is currently:

Sepia slimes now spawn "Bluespace Stopclocks" when the extract is given plasma, "Strangely fluctuating weight, its constant shifting in space causes it to take up more room than you'd think. Halts time on activation in a small area". After creation they take a minute to wind up before they're able to be used, and placing them in a bluespace bag will cause a tear in reality the same as placing a bluespace bag in another bluespace bag. These take up a "laser gun size" space in a bag severely limiting how many can be carried and stops rapid spam creating and using of them due to their prep time after creation.

This will mean a player is very limited on how many they can carry, they sacrifice being able to have a bluspace bag, and if they do carry them then they're sacrificing a lot space in their bag that could otherwise be used by crossbreeds, weapons and other. Feedback would be appreciated.

**Comment by dylanstrategie** [comment]
Yes brother, bring on the nerfs. Teach them

**Comment by wesoda25** [comment]
> Been talking about balancing this a lot in discord, main problem is that you can effectively carry tons of these making them disproportionately strong for how easy they are to get. A single use sepia would be less deadly than a grenade, and you have to work far harder to get it than you would to even get a maxcap. It's the botany problem again, the sheer volume of a certain strong weapon you can carry thanks to department plant/bio bags. Spamming timestops, making you effectively unstoppable is the issue here.
> 
> So the **proposed solution** which we like as a basepoint nerf to try is currently:
> 
> Sepia slimes now spawn "Bluespace Stopclocks" when the extract is given plasma, "Strangely fluctuating weight, its constant shifting in space causes it to take up more room than you'd think. Halts time on activation in a small area". After creation they take a minute to wind up before they're able to be used, and placing them in a bluespace bag will cause a tear in reality the same as placing a bluespace bag in another bluespace bag. These take up a "laser gun size" space in a bag severely limiting how many can be carried and stops rapid spam creating and using of them due to their prep time after creation.
> 
> This will mean a player is very limited on how many they can carry, they sacrifice being able to have a bluspace bag, and if they do carry them then they're sacrificing a lot space in their bag that could otherwise be used by crossbreeds, weapons and other. Feedback would be appreciated.

Make a PR and force the feedback.

**Comment by ExcessiveUseOfCobblestone** [comment]
The core issue for me is that it gives the scientist immunity to the field. For wizard it makes sense I guess because lol muh 1v50 but certainly not for a science item.

**Comment by Shardj** [comment]
> The core issue for me is that it gives the scientist immunity to the field. For wizard it makes sense I guess because lol muh 1v50 but certainly not for a science item.

Do you not think it'd be useless without that affect? Also I really don't think it's any more powerful than a regular grenade, bomb,  any botany bullshit or any of the rest when compared as a single use item. And it requires far more work to acquire. than the above mentioned items. However, if the bluespace stopclock still isn't enough of a nerf things can be tweaked, honestly though after that I can't see people calling sepia overpowered when compared to gold mobs or adamantium.

**Comment by ExcessiveUseOfCobblestone** [comment]
No that's why I think it's the core issue lol. It's why it needs to be delayed so you can drop it, it WOULD be useless if it was instant and 100% trapped you inside.

**Comment by ExcessiveUseOfCobblestone** [comment]
Also any other BS xenobio can obtain does not justify an arms race with itself.

**Comment by Shardj** [comment]
You can only throw stuff then to affect the person trapped, xenobio doesn't synergise with that. It also makes all of the sepia crossbreeds useless.

**Comment by 4dplanner** [comment]
It doesn't make ALL of the crossbreeds useless

**Comment by Shardj** [comment]
> It doesn't make ALL of the crossbreeds useless

It makes all of the ones that aren't already terrible terrible too

**Comment by Shardj** [comment]
> Yes it's almost like timestop should be used to get out of an area instead of allowing for free hits on everyone inside with no counterplay.

There is no should, it's whatever it's chosen to be. And that hasn't been its purpose in the past so you can't really pull that. However if you feel so strongly about this, then I recommend you take this nerf as it is on this branch and merge it. But I know I and others aren't happy with the state of this and other nerfs recently.

**Comment by optimumtact** [comment]
oh no, whatever will we do to restore shardj's happiness.

**Comment by Shardj** [comment]
> oh no, whatever will we do to restore shardj's happiness.

You act like I didn't just say you can merge this, I'm just a peon, feel free. I'm not dramatic enough to claim tg is dead like some people because of minor nerfs.

**Comment by KomradeSpectre** [comment]
> Yes it's almost like timestop should be used to get out of an area instead of allowing for free hits on everyone inside with no counterplay.

Making it a huge slowdown instead of a complete stop would be good for adding counterplay, but retaining it's strength. Everyone inside of the field would be moving slower(we're talking like Citrus tier slow), and the projectiles would be moving slow. Some other changes I can't really think of would have to be made to make this viable.

**Comment by 4dplanner** [comment]
If shardj's solution isn't acceptable, I have thought up a solution to the recurring thing: have it recall to hand after activating (or possibly after the effect ends)

**Comment by OverlyGeneric** [comment]
> So the **proposed solution** which we like as a basepoint nerf to try is currently:
> 
> Sepia slimes now spawn "Bluespace Stopclocks" when the extract is given plasma, "Strangely fluctuating weight, it's constant shifting in space causes it to take up more room than you'd think. Halts time on activation in a small area". After creation they take a minute to wind up before they're able to be used, and placing them in a bluespace bag will cause a tear in reality the same as placing a bluespace bag in another bluespace bag. These take up a "laser gun size" space in a bag severely limiting how many can be carried and stops rapid spam creating and using of them due to their prep time after creation.
> 
> This will mean a player is very limited on how many they can carry, they sacrifice being able to have a bluspace bag, and if they do carry them then they're sacrificing a lot space in their bag that could otherwise be used by crossbreeds, weapons and other. Feedback would be appreciated.

Holy moly, this sounds fantastic. 4dplanner why don't you put some extra effort into your PR and make a feature instead of a removal?

This is an interesting and clever concept that would work. **Steal the idea** and make it your own. It's free real estate.


> Next thing you know xenobio will be giving themselves fireball and bolt of death because of the “fun” argument.

**No, do not. Bad soda. Do NOT GIVE HIM IDEAS.**

**Comment by Tlaltecuhtli** [comment]
what if instead it stopped only items(projectiles, trowns,etc) and made the people in the field slown down? so they could have an anime fight

**Comment by ExcessiveUseOfCobblestone** [comment]
@KomradeSpectre idc what happens in the field as long as the spawner is affected by it as well. Getting free hits whether it's because they literally can't do anything or because they're too slow to strike back isn't fun.

**Comment by Shardj** [comment]
@ExcessiveUseOfCobblestone you're gonna hate these obscure things called tazers.

**Comment by ExcessiveUseOfCobblestone** [comment]
There are several ways to avoid and/or mitigate stuns and it's quite obvious when they have tasers out.

**Comment by Shardj** [comment]
@ExcessiveUseOfCobblestone they're also a round start item. If you want to be serious about it though. How do you plan to deal with a guy who has a bluespace bag full of IED spears. I ask because in my mind they're equally powerful and neither have good counterplay apart from the fact that you can carry a fuck ton of sepias but only a few spears. I'll quite often go for grenade spears in a bluespace bag along with some helpful stabilized crossbreeds.

I can honestly see what you're getting at, I just don't view it as a problem.

**Comment by ExcessiveUseOfCobblestone** [comment]
I've already said in another PR I would love to see those removed as well.

**Comment by Shardj** [comment]
I'm looking forward to the drama of that one, pretty much the only positive thing about it though

**Comment by OverlyGeneric** [comment]
> Sepia slimes now spawn "Bluespace Stopclocks" when the extract is given plasma, "Strangely fluctuating weight, it's constant shifting in space causes it to take up more room than you'd think. Halts time on activation in a small area". After creation they take a minute to wind up before they're able to be used, and placing them in a bluespace bag will cause a tear, in reality, the same as placing a bluespace bag in another bluespace bag. These take up a "laser gun size" space in a bag severely limiting how many can be carried and stops rapid spam creating and using of them due to their prep time after creation.


Sounds cool, 4dplanner put some extra effort into your PR and implement this. It's an interesting and clever concept.

**Comment by 4dplanner** [comment]
Was unironically in talks with shardj about this, and he said he'd code it himself.

**Comment by ExcessiveUseOfCobblestone** [comment]
When are we replacing timestop with wizard-time so I can summon Gomorrah to kill the colossus?

**Comment by wesoda25** [comment]
Tasers are stupid (kinda eh) but theres a ton of counterplay to them. Anti magic doesn’t even work now so you’re a God with sepia extract.

**Comment by Shardj** [comment]
> Was unironically in talks with shardj about this, and he said he'd code it himself.

I'm only going to do it if maintainers don't shoot it down, I would've already started if I hadn't received such negative reactions from oranges and cobble. I've heard positive responses from most others so far though.

**Comment by Shardj** [comment]
> Tasers are stupid (kinda eh) but theres a ton of counterplay to them. Anti magic doesn’t even work now so you’re a God with sepia extract.

Getting in a regular stun or grenade is still just as effective. You become a station wide threat yes, are you unstoppable, not at all. But yeah it's too strong for how easy it is to get tons of.

**Comment by Shardj** [comment]
https://www.piskelapp.com/p/agxzfnBpc2tlbC1hcHByEwsSBlBpc2tlbBiAgKC4p-D2Cgw/view

I only shaded the first frame properly for now as it probably won't be animated except maybe at activation? But that's a rough idea of how I imagine it

**Comment by wesoda25** [comment]
Wait do you intend for the animation to begin once activated (plasma injected), and once it finishes the animation the clock stops and time actually freezes? That'd be really cool.

**Comment by Shardj** [comment]
> Wait do you intend for the animation to begin once activated (plasma injected), and once it finishes the animation the clock stops and time actually freezes? That'd be really cool.

I have no plans for how to use the animations, I haven't thought about it

**Comment by ExcessiveUseOfCobblestone** [comment]
IDC how it's approached as long as the end product makes the user, by default at least, susceptible to the field.

**Comment by ExcessiveUseOfCobblestone** [comment]
Update on this btw?

**Comment by Shardj** [comment]
If that's your unmoving position then you'll never be happy with my nerf, so may as well merge this in as it currently is @ExcessiveUseOfCobblestone as I can't see a better solution.

**Comment by Shardj** [comment]
@4dplanner any thoughts?

**Comment by Shardj** [comment]
Hm, another idea, merge this in as it currently is. But maybe stablized sepia gives you partial immunity to the time field? The current stablized sepia is terrible

**Comment by 4dplanner** [comment]
I'm still not looking at this until things cool down a bit IRL, but I'll re-open later if shardj's solution doesn't go through.

**Comment by 4dplanner** [comment]
here we go again

**Comment by Whoneedspacee** [comment]
ok

**Comment by 4dplanner** [comment]
Please can I direct you to my new website at https://github.com/tgstation/tgstation/pull/42933

**Comment by wesoda25** [comment]
interesting

**Pull request closed without merging**